internal/querycoord/meta.go

```diff
@@ -647,10 +644,12 @@ func (m *MetaReplica) saveGlobalSealedSegInfos(saves col2SegmentInfos) (col2Seal
 		}
 	}
 
-	// remove compacted segment info from etcd
-	for _, segmentInfo := range segmentsCompactionFrom {
-		if err := m.segmentsInfo.removeSegment(segmentInfo); err != nil {
-			panic(err)
+	// remove segmentInfos to remove
+	for _, infos := range removes {
+		for _, info := range infos {
```

Comment:
Now removes will only be generated when handoff. If the saveGlobalSealedSegInfos function is used incorrectly, such as in the load balance scenario, the segment meta will be incorrectly cleaned up. It is best to add a comment to this place.

Reply:
Ac

> Now removes will only be generated when handoff. If the saveGlobalSealedSegInfos function is used incorrectly, such as in the load balance scenario, the segment meta will be incorrectly cleaned up. It is best to add a comment to this place.

For loadbalance scenario, the release should be handled by saveGlobalSealedSegInfos itself because the do a diff operation to find previous online list and current online list.

Reply:
It's still a trick but I can't find a good way to do so without refactoring the code